src/tests/test_utils/unit_test_utils/mocks/mock_engine/mock_extensions.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/core/op_extension.hpp"
+#include "openvino/op/op.hpp"
+
+class MockOp : public ov::op::Op {
+public:
+    OPENVINO_OP("MockOp", "custom_opset");
+    MockOp() = default;
+
+    bool visit_attributes(ov::AttributeVisitor& visitor) override {
+        return true;
+    }
+
+    void validate_and_infer_types() override {
+        set_output_size(1);
+        set_output_type(0, get_input_element_type(0), get_input_partial_shape(0));
+    }
+
+    std::shared_ptr<Node> clone_with_new_inputs(const ov::OutputVector& inputs) const override {
+        return nullptr;
```

Comment:
Optional:
Throw instead NULL pointer, usually after clone the pointer usually is not validated as it will create valid node.